### PR TITLE
Compare last modified time in ms.

### LINKED ISSUE #574: Test failures on Mac OS
```
[ERROR] Failures:
[ERROR]   XMLSchemaCompletionExtensionsTest.completionWithXMLSchemaContentChanged:396 expected:<4> but was:<5>
[ERROR]   XMLSchemaCompletionExtensionsTest.xsAny:752 expected:<8> but was:<6>
[ERROR]   XMLValidationPoolCacheTest.dtd:240->testDiagnosticsFor:246 Unexpected diagnostics:
[] expected:<[Diagnostic [
  range = Range [
    start = Position [
      line = 2
      character = 1
    ]
    end = Position [
      line = 2
      character = 5
    ]
  ]
  severity = null
  code = "MSG_CONTENT_INCOMPLETE"
  source = null
  message = ""
  relatedInformation = null
]]> but was:<[]>
[ERROR]   XMLValidationPoolCacheTest.includedSchemaLocation:139->testDiagnosticsFor:246 Unexpected diagnostics:
[Diagnostic [
  range = Range [
    start = Position [
      line = 2
      character = 2
    ]
    end = Position [
      line = 2
      character = 5
    ]
  ]
  severity = Error
  code = "cvc-complex-type.2.4.b"
  source = "xml"
  message = "Child elements are missing from element:\n - Bar\n\nThe following elements are expected:\n - FooBar\n\nError indicated by\n {the schema}\nwith code:"
  relatedInformation = null
]] expected:<[Diagnostic [
  range = Range [
    start = Position [
      line = 2
      character = 6
    ]
    end = Position [
      line = 4
      character = 1
    ]
  ]
  severity = null
  code = "cvc-complex-type.2.1"
  source = null
  message = ""
  relatedInformation = null
]]> but was:<[Diagnostic [
  range = Range [
    start = Position [
      line = 2
      character = 2
    ]
    end = Position [
      line = 2
      character = 5
    ]
  ]
  severity = null
  code = "cvc-complex-type.2.4.b"
  source = null
  message = ""
  relatedInformation = null
]]>
[INFO]
[ERROR] Tests run: 802, Failures: 4, Errors: 0, Skipped: 3
```

### PR DESCRIPTION
**Pull request by angelozerr**
Compare last modified time in ms.

Fixes #574

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
@fbricon could you test this PR in Mac OS please to check if all tests are done with success.

**Comment by angelozerr** [comment]
@fbricon don't check it, it fails again -(

**Comment by fbricon** [comment]
```
[ERROR] Failures:
[ERROR]   XMLSchemaCompletionExtensionsTest.completionWithXMLSchemaContentChanged:397 expected:<4> but was:<5>
[ERROR]   XMLSchemaCompletionExtensionsTest.xsAny:753 expected:<8> but was:<6>
[ERROR]   XMLValidationPoolCacheTest.dtd:240->testDiagnosticsFor:246 Unexpected diagnostics:
[] expected:<[Diagnostic [
  range = Range [
    start = Position [
      line = 2
      character = 1
    ]
    end = Position [
      line = 2
      character = 5
    ]
  ]
  severity = null
  code = "MSG_CONTENT_INCOMPLETE"
  source = null
  message = ""
  relatedInformation = null
]]> but was:<[]>
[ERROR]   XMLValidationPoolCacheTest.includedSchemaLocation:139->testDiagnosticsFor:246 Unexpected diagnostics:
[Diagnostic [
  range = Range [
    start = Position [
      line = 2
      character = 2
    ]
    end = Position [
      line = 2
      character = 5
    ]
  ]
  severity = Error
  code = "cvc-complex-type.2.4.b"
  source = "xml"
  message = "Child elements are missing from element:\n - Bar\n\nThe following elements are expected:\n - FooBar\n\nError indicated by\n {the schema}\nwith code:"
  relatedInformation = null
]] expected:<[Diagnostic [
  range = Range [
    start = Position [
      line = 2
      character = 6
    ]
    end = Position [
      line = 4
      character = 1
    ]
  ]
  severity = null
  code = "cvc-complex-type.2.1"
  source = null
  message = ""
  relatedInformation = null
]]> but was:<[Diagnostic [
  range = Range [
    start = Position [
      line = 2
      character = 2
    ]
    end = Position [
      line = 2
      character = 5
    ]
  ]
  severity = null
  code = "cvc-complex-type.2.4.b"
  source = null
  message = ""
  relatedInformation = null
]]>
[ERROR]   FilesChangedTrackerTest.trackFile:21
[ERROR] Errors:
[ERROR]   XMLExternalTest.externalXSDTest:150 » IndexOutOfBounds Index: 0, Size: 0
[INFO]
[ERROR] Tests run: 803, Failures: 5, Errors: 1, Skipped: 3
```

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/model/FilesChangedTracker.java [hunk 1]
```diff
@@ -54,7 +55,8 @@ public boolean isDirty() {
 					return true;
 				}
 				FileTime currentLastMofied = Files.getLastModifiedTime(file);
-				if (!currentLastMofied.equals(lastModified)) {
+				if (lastModified == null
+						|| currentLastMofied.to(TimeUnit.MICROSECONDS) != lastModified.to(TimeUnit.MICROSECONDS)) {
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/model/FilesChangedTracker.java]: compare using equals

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/model/FilesChangedTracker.java [hunk 2]
```diff
@@ -54,7 +55,8 @@ public boolean isDirty() {
 					return true;
 				}
 				FileTime currentLastMofied = Files.getLastModifiedTime(file);
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/model/FilesChangedTracker.java]: currentLastModified

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/extensions/contentmodel/BaseFileTempTest.java [hunk 3]
```diff
@@ -42,7 +45,28 @@ private static void createTempDir() {
...
+			Files.write(path, contents.getBytes(), StandardOpenOption.CREATE, StandardOpenOption.WRITE);
+			FileTime time = FileTime.fromMillis(System.currentTimeMillis());
+			Files.setLastModifiedTime(path, time);
+		} catch (URISyntaxException e) {
+			e.printStackTrace();
+		}
+	}
+
+	protected static void updateFile(String fileName, String contents) throws IOException {
+		createFile(fileName, contents);
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/extensions/contentmodel/BaseFileTempTest.java]: the file should be modified after the wait period, else it'll be too fast

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/extensions/contentmodel/BaseFileTempTest.java [hunk 4]
```diff
@@ -42,7 +45,28 @@ private static void createTempDir() {
...
+		try {
+			URI uri = new URI("file://" + fileName);
+			Path path = Paths.get(uri);
+			Files.write(path, contents.getBytes(), StandardOpenOption.CREATE, StandardOpenOption.WRITE);
+			FileTime time = FileTime.fromMillis(System.currentTimeMillis());
+			Files.setLastModifiedTime(path, time);
+		} catch (URISyntaxException e) {
+			e.printStackTrace();
+		}
+	}
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/extensions/contentmodel/BaseFileTempTest.java]: change not necessary

**Pull request merged**